### ISSUE TITLE
[BUG] Double rendering Latex artifact in titles

### ISSUE THREAD

**Issue by peekxc** [bokeh/bokeh#14129]
### Software versions

Python version        :  3.11.0 (main, Mar  1 2023, 12:49:28) [Clang 14.0.6 ]
IPython version       :  8.25.0
Tornado version       :  6.4.1
Bokeh version         :  3.5.1
BokehJS static path   :  /Users/mpiekenbrock/opt/miniconda3/envs/spirit/lib/python3.11/site-packages/bokeh/server/static
node.js version       :  v22.3.0
npm version           :  10.8.1
jupyter_bokeh version :  (not installed)
Operating system      :  macOS-10.16-x86_64-i386-64bit

### Browser name and version

Chrome Version 129.0.6668.101 (Official Build) (x86_64)

### Jupyter notebook / Jupyter Lab version

NA

### Expected behavior

Only a single render should occur

### Observed behavior

No matter what version of Bokeh I run, I observe all latex labels as double rendered

### Example code

```Python
import numpy as np
from bokeh.plotting import figure, show

x = np.linspace(0, 2*np.pi, 100)
p = figure(width=400, height=200, title=r"$$\cos(3 * \tau)$$")
p.line(x, np.cos(3*x))
show(p)
```

### Screenshots

Below is the exact SVG rendered by bokeh in the browser with the artifacts

![bad_equation](https://github.com/user-attachments/assets/4a239ebd-f0f9-4540-9e5a-414743a608fa)

Also below is a screen shot of another equation as rendered in the browser (or jupyter notebook). 

<img width="747" alt="Screenshot 2024-10-29 at 4 28 40 PM" src="https://github.com/user-attachments/assets/1ebe8845-6ff6-432b-a32c-7497e7d50259">

The behavior has persisted across several bokeh versions for my system.

**Comment by bryevdv** [comment]
@peekxc Do you see the issue on this page? 

https://docs.bokeh.org/en/latest/docs/examples/styling/mathtext/latex_blackbody_radiation.html

I do not see any issue there on OXS/Chrome. 

Does the issue only happen with `title` or have you see this happen with math text in other places? The latest release is 3.6, not 3.5.1 do you see the issue with 3.6?

**Comment by bryevdv** [comment]
I do see the issue with the MRE and I also see it here, but only in the title: 

https://docs.bokeh.org/en/latest/docs/examples/styling/mathtext/latex_bessel.html

@mattpap I think this problem is specific to titles, at least I don't see it on labels, axis labels, or DOM elements.

**Comment by mattpap** [comment]
This may be a bug in MathJax or in the font. `Title` uses bold font by default, which seems to be the cause. If this line:
```py
p.title.text_font_style = "normal"
```
is added to the MRE, then `Title` is rendered normally:

![image](https://github.com/user-attachments/assets/861d2ed6-79a8-4c5a-af6e-38b43038bc34)

**Comment by mattpap** [comment]
Also, I verified that bokehjs is painting as expected and the double image comes from MathJax.

**Comment by bryevdv** [comment]
Actually I now seem to have a vague recollection of some issue specifically with math text and bold from the past. I will have to try to search around for it later though.

Edit: maybe this https://github.com/bokeh/bokeh/issues/12531

Maybe we just give up on making the title bold if math text is detected

**Comment by peekxc** [comment]
> ```python
> p.title.text_font_style = "normal"
> ```

I can confirm this works as an alternative! 

> @peekxc Do you see the issue on this page
>
> https://docs.bokeh.org/en/latest/docs/examples/styling/mathtext/latex_blackbody_radiation.html

Actually I do not. 

> Does the issue only happen with title or have you see this happen with math text in other places? The latest release is 3.6, not 3.5.1 do you see the issue with 3.6?

Also, yes, it is only the title element. I will have to check if it works with 3.6.

---

Inspecting the actual SVG after downloading, I can confirm there are two actual MathJax elements:

<img width="553" alt="Screenshot 2024-10-29 at 6 20 42 PM" src="https://github.com/user-attachments/assets/cb584fcc-0ff1-453c-b78f-8d3b6cba36a6">

Removing the second element (or even just removing the translation) fixes the issue, but I don't see how to do this from the bokeh side.

**Comment by bryevdv** [comment]
Well, AFAIK we didn't actually ever try the [`boldsymbol`](https://docs.mathjax.org/en/latest/input/tex/extensions/boldsymbol.html) option since it seemed like the simple, built-in  `\pmb` command from AMS was sufficient. But I think that's the only option left (MathJax still does not support `bm` that I am aware). So if `boldsymbol` does not work then I do think maybe we just have to punt on the combination of bold and mathtext, at least for now, unless we want to try to do ugly post-facto surgery on MathJax output (seems like a bad idea to me)

**Comment by bryevdv** [comment]
@mattpap can you comment how the instructions at [boldsymbol](https://docs.mathjax.org/en/latest/input/tex/extensions/boldsymbol.html) might be applied to our codebase? Is this a modification that can be applied in [`providers.ts`](https://github.com/bokeh/bokeh/blob/branch-3.7/bokehjs/src/lib/models/text/providers.ts)?

**Comment by bryevdv** [comment]
Actually, maybe `boldsymbol` just works out of the box now? 

With this diff
```diff
diff --git a/bokehjs/src/lib/models/text/math_text.ts b/bokehjs/src/lib/models/text/math_text.ts
index 33c8cb1488..a59e980aee 100644
--- a/bokehjs/src/lib/models/text/math_text.ts
+++ b/bokehjs/src/lib/models/text/math_text.ts
@@ -555,7 +555,7 @@ export class TeXView extends MathTextView {
   override get styled_text(): string {
     const [r, g, b] = color2rgba(this.color)

-    return `\\color[RGB]{${r}, ${g}, ${b}} ${this.font.includes("bold") ? `\\pmb{${this.text}}` : this.text}`
+    return `\\color[RGB]{${r}, ${g}, ${b}} ${this.font.includes("bold") ? `\\boldsymbol{${this.text}}` : this.text}`
   }

   protected _process_text(): HTMLElement | undefined {
```

things already look improved on chrome:

### Before (smudged)

<img width="1019" alt="Image" src="https://github.com/user-attachments/assets/35252a8a-d11c-4157-8b03-428bc4ad93ed" />


### After (clear)

<img width="821" alt="Image" src="https://github.com/user-attachments/assets/1400a57d-4201-48fb-a784-0f10d6b20fae" />
